### PR TITLE
Breath mask auto-adjust

### LINKED ISSUE #7276: Misleading feedback on internals status
Trying to toggle on your internals with the button while your breath mask is down will do the regular message and change the icon to be on, only to turn off a few seconds later without even telling you.

http://i.imgur.com/QmWbNBG.png

It should either not allow it (and tell you that your breath mask needs to be lifted up) or do that for you and automatically strap it back on.

### PR DESCRIPTION
**Pull request by kingofkosmos**
Now if you have a breath mask on your face, but lowered, and decide to start running on internals, you will automatically adjust the mask back on your face!

Also you're able to adjust your mask now if you're buckled to a chair, which you couldn't do before.

Fixes #7276

### CODE REVIEW

**Comment by RemieRichards** [comment]
Smart :+1:

File: code/_onclick/hud/screen_objects.dm [hunk 1]
```diff
@@ -126,13 +126,16 @@
...
 				icon_state = "internal0"
 			else
 				if(!istype(C.wear_mask, /obj/item/clothing/mask))
-					C << "<span class='notice'>You are not wearing a mask.</span>"
+					C << "<span class='warning'>You are not wearing a mask!</span>"
 					return 1
 				else
+					var/obj/item/clothing/mask/M = C.wear_mask
+					if(M.mask_adjusted) // if mask on face but pushed down
+						M.adjustmask(usr) // automatically adjust it back
```
Comment (ComicIronic) [inline code/_onclick/hud/screen_objects.dm]: Why not use (C) instead of (usr)?

**Comment by WJohn** [comment]
:+1:

File: code/_onclick/hud/screen_objects.dm (hunk 1 above)
Comment (RemieRichards) [inline code/_onclick/hud/screen_objects.dm]: Yeah @kingofkosmos Comic's right, sort this out, usr isn't as safe as C.

**Comment by ACCount12** [comment]
:+1: Now adjusting masks is useable.

**Comment by xxalpha** [comment]
You could also make so when you manually turn off internals the mask is lowered.

**Comment by WJohn** [comment]
Makes me wonder why we even have a toggle then and not be able to make it automatic. If it's on it's on your face, if it's off it's around your chin so you can eat.

**Comment by RandomMarine** [comment]
I'm with WJohn's suggestion. Fights action button bloat? Visual feedback for players on who's currently using internals? I like that.

If somebody wants to hide whether or not they have internals on, they can still find a gas mask in maint or something.

**Comment by ComicIronic** [comment]
The issue with auto-lowering it is that someone might want mouth protection from a mask without having internals.
This is only good because an in-place mask is a requirement. Not having an in-place mask is no requirement for breathing normally.

**Comment by WJohn** [comment]
The thing is that we don't have any other mask we can toggle with actual functional differences like that. Gas masks can't be toggled on/off your face for toxin filtering properties, all the other masks have no effects, and I can't even think of any mask (maybe bandanas) that can be toggled (which still don't have any properties for being on/off).

**Comment by Lumipharon** [comment]
Bandana's can't be toggled, why were they ported from NT without the toggle?

**Comment by MrPerson** [comment]
Mouth protection isn't a good reason anyway

**Comment by Alek2ander** [comment]
Security gas masks though

**Comment by kingofkosmos** [comment]
I would make the breath masks adjust automatically up/down always and remove the button if I knew how... But even then I wouldn't know what to do with the doctor's sterile masks and other possible masks.

And some people possibly like to have the breath mask up to protect against forcefeeding and airborne diseases.

**Comment by Lumipharon** [comment]
Just stick with the original idea, it works well.

File: code/_onclick/hud/screen_objects.dm [hunk 2]
```diff
@@ -126,13 +126,16 @@
...
-				C << "<span class='notice'>No longer running on internals.</span>"
+				C << "<span class='notice'>You are no longer running on internals.</span>"
 				icon_state = "internal0"
 			else
 				if(!istype(C.wear_mask, /obj/item/clothing/mask))
-					C << "<span class='notice'>You are not wearing a mask.</span>"
+					C << "<span class='warning'>You are not wearing a mask!</span>"
 					return 1
 				else
+					var/obj/item/clothing/mask/M = C.wear_mask
```
Comment (phil235) [inline code/_onclick/hud/screen_objects.dm]: You could fix  7276 and 8771 by adding:
`if( !(wear_mask.flags & MASKINTERNALS) )`
`  C << "<span class='warning'>You need an internals mask blahblahblah !</span>"`
`  return`

Comment (kingofkosmos) [inline code/_onclick/hud/screen_objects.dm]: I tried that, but now it says "You need an internals mask blahblahblah !" when I try to turn internals on with the breath mask adjusted down.

Comment (phil235) [inline code/_onclick/hud/screen_objects.dm]: In that case, move it after `if(M.mask_adjusted) M.adjustmask(C)`

**Comment by kingofkosmos** [comment]
Ok good to go! Thanks Phil235 for all the help.

**Pull request merged**